src/poetry/utils/env/base_env.py

```diff
@@ -242,6 +244,41 @@ def paths(self) -> dict[str, str]:
 
         return self._paths
 
+    @property
```

Comment:
We may use `@cached_property` instead so we don't need `_installer_scheme_dict`.